components/automate-ui/src/app/page-components/simple-line-graph/simple-line-graph.component.ts

```diff
@@ -0,0 +1,342 @@
+  get rangeY(): [number, number] {
+    const min = this.margin.top;
+    const max = this.height - this.margin.bottom;
+    return [max, min];
+  }
+
+  get domainY(): [number, number] {
+    const min = 0;
+    const max = 100; // since this based on a percentage we are doing 0 to 100;
+    return [min, max];
+  }
+
+  get yScale() {
+    return d3.scaleLinear()
+             .range(this.rangeY)
+             .domain(this.domainY);
+  }
+
+  //////// SELECTIONS ////////
+  get containerSelection() {
+    return d3.select('app-simple-line-graph');
+  }
+
+  get svgSelection() {
+    return d3.select(this.svg.nativeElement);
+  }
+
+  get axisYSelection() {
+    return this.svgSelection.select('.y-axis');
+  }
+
+  // returns a function that when passed our data, will return an svg path
+  get createPath() {
+    return d3.line()
+              .x(d => this.xScale( d.daysAgo) )
+              .y(d => this.yScale( d.percentage) );
+  }
+
+  get viewBox() {
+    return `0 0 ${this.width} ${this.height}`;
+  }
+
+    ////////////////// RENDER FUNCTIONS ////////////////////
+  private renderChart() {
+    this.AllUnlockDeactivate();
+    this.resizeChart();
+    this.renderGrid();
+    this.renderLine();
+    this.renderPoints();
+    this.renderTooltips();
+    this.renderRings();
+    this.renderLabelButtons();
+    this.relock();
+  }
+
+  private renderLine(): void {
+    // create the line using path function
+    const line = this.createPath(this.data);
+
+    const theLine = this.svgSelection.selectAll('.line').data([this.data], d => d.daysAgo);
+    theLine.exit().remove();
+    theLine.enter().append('path').attr('class', 'line').merge(theLine)
+      .transition().duration(1000)
+    .attr('d', line);
+  }
+
+  private renderPoints(): void {
+    const points = this.svgSelection.selectAll('circle.point')
+      .data(this.data, d => d.daysAgo);
+    points.exit().remove();
+    points.enter().append('circle')
+        .attr('class', (_d, i) => `point elem-${i}`)
+        .merge(points)
+        .transition().duration(1000)
+        .attr('percent', ( d => d.percentage ) )
+        .attr('cx', d => this.xScale(d.daysAgo))
+        .attr('cy', d => this.yScale(d.percentage))
+        .attr('r', 4);
+  }
+
+  private renderRings(): void {
+    const rings = this.svgSelection.selectAll('circle.ring')
+      .data(this.data, d => d.daysAgo);
+    rings.exit().remove();
+    rings.enter().append('circle')
+      .attr('class', (_d, i) => `ring elem-${i}`)
+      .merge(rings)
+      .attr('cx', d => this.xScale(d.daysAgo))
+      .attr('cy', d => this.yScale(d.percentage))
+      .attr('r', 10);
+  }
+
+  private renderTooltips(): void {
+    // these numbers are specific to its container
+    const localWidth = this.width - this.margin.right - this.margin.left - 34;
+    const thisRange = [localWidth, -this.margin.right];
+    const thisScale = d3.scaleLinear()
+      .domain(this.domainX)
+      .range(thisRange);
+
```

Comment:
the tooltips have been a bit difficult to work with.  The reason is that we cannot append html elements to an SVG, instead we have to attach them somewhere in the dom.  If we attach to the body, the positioning is easier, but we have unintended scroll issues, i.e. a tooltip scrolls with us when it shouldnt.
Instead I've chosen to attach it to the svg parent container.  The best thing I could come up with to make the scaling fit across the X axis is to adjust the scaling down to roughly what the svg container is.